sympy/solvers/ode/single.py

```
@@ -359,6 +358,137 @@ def inverse(self):
+                    factor = simplify(numerator/n)
+                    variables = factor.free_symbols
+
+                    if len(variables) == 1 and x == variables.pop():
+                        factor = exp(Integral(factor).doit())
+                        m*= factor
+                        n*= factor
+                        self.p=m.subs(y,f)
+                        self.q=n.subs(y,f)
+                        self._matched=True
+                        return True
+                    else:
+                        # If (dP/dy - dQ/dx) / -P = f(y)
+                        # then exp(integral(f(y))*equation becomes exact
+                        factor = simplify(-numerator/m)
+                        variables = factor.free_symbols
+                        if len(variables) == 1 and y == variables.pop():
+                            factor = exp(Integral(factor).doit())
+                            m*= factor
+                            n*= factor
+                            self.p=m.subs(y,f)
+                            self.q=n.subs(y,f)
+                            self._matched=True
+                            return True
+                else:
+                    self._matched=True
+                    return True
+        except NotImplementedError:
+            # Differentiating the coefficients might fail because of things
+            # like f(2*x).diff(x).  See issue 4624 and issue 4719.
+            pass
+
+        self._matched=False
+        return False
+
+    def _get_general_solution(self, *, simplify: bool = True) -> List[Equality]:
+        y=Dummy('y')
+        f=self.ode_problem.func
+        x=self.ode_problem.sym
+        (C1,) = self.ode_problem.get_numbered_constants(num=1)
+
+        m=self.p.subs(f,y)
+        n=self.q.subs(f,y)
+
+        sol= Integral(m,x)+Integral(n-Integral(m,x).diff(y),y)
+
+        try:
+            sol=sol.doit()
+            gen_sol=Eq(sol.subs(y,f),C1)
+        except:
```

Bare except should not be used anywhere in the codebase

---

Removing `try except` block as `doit` will not be called here.